scripts/lua/management/lib/swagger.lua

```diff
@@ -0,0 +1,185 @@
+--   FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+--   AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+--   LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+--   FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+--   DEALINGS IN THE SOFTWARE.
+
+--- @module swagger
+-- Module for parsing swagger file
+
+local _M = {}
+
+-- Convert passed-in swagger body to valid lua table
+-- @param swagger swagger file to parse
+function _M.parseSwagger(swagger)
+  local backends = parseBackends(swagger)
+  local policies = parsePolicies(swagger)
+  local security = parseSecurity(swagger)
+  local decoded = {
+    name = swagger.info.title,
+    basePath = swagger.basePath,
+    resources = {}
+  }
+  for path, verbObj in pairs(swagger.paths) do
+    decoded.resources[path] = { operations = {} }
+    for verb, value in pairs(verbObj) do
+      decoded.resources[path].operations[verb] = {}
+      local verbObj = decoded.resources[path].operations[verb]
+      local backend = (backends["all"] ~= nil) and backends["all"] or backends[value.operationId]
+      verbObj.backendUrl = backend.backendUrl
+      verbObj.backendMethod = (backend.backendMethod == 'keep') and verb or backend.backendMethod
+      verbObj.policies = policies
+      verbObj.security = security
+    end
+  end
+  return decoded
+end
+
+--- Parse backendUrl and backendMethod
+-- @param swagger swagger file to parse
+function parseBackends(swagger)
+  local configObj = swagger["x-gateway-configuration"]
+  configObj = (configObj == nil) and swagger["x-ibm-configuration"] or configObj
+  if configObj ~= nil then
+    for _, obj in pairs(configObj.assembly.execute) do
+      for policy, v in pairs(obj) do
+        local res = {}
+        if policy == "operation-switch" then
+          local caseObj = v.case
+          for _, case in pairs(caseObj) do
+            for _, op in pairs(case.operations) do
+              res[op] = {
+                backendUrl = case.execute[1]["proxy"]["target-url"],
+                backendMethod = case.execute[1]["proxy"].verb
+              }
+            end
+          end
+          return res
+        end
+        if policy == "proxy" then
+          res["all"] = {
+            backendUrl = v["target-url"],
+            backendMethod = v.verb
+          }
+          return res
+        end
+      end
+    end
+  end
+end
+
+--- Parse policies in swagger
+-- @param swagger swagger file to parse
+function parsePolicies(swagger)
+  local policies = {}
+  -- parse rate limit
+  policies = parseRateLimit(swagger, policies)
+  policies = parseRequestMapping(swagger, policies)
+  return policies
+end
+
+--- Parse rate limit
+function parseRateLimit(swagger, policies)
+  local rlObj = swagger["x-gateway-rate-limit"]
+  rlObj = (rlObj == nil) and swagger["x-ibm-rate-limit"] or rlObj
+  if rlObj ~= nil then
+    rlObj = rlObj[1]
+    if rlObj.unit == "second" then
```

Comment:
Probably want to default to one if these if we don't understand the unit sent in since it doesn't look like we're failing it.